### ISSUE TITLE
linter: `unicorn/consistent-function-scoping` ignores outer bindings used as JSX element names

### ISSUE THREAD

**Issue by lukpsaxo** [oxc-project/oxc#26592]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
    "plugins": ["unicorn"],
    "rules": {
        "unicorn/consistent-function-scoping": [
            "error",
            { "checkArrowFunctions": false }
        ]
    }
}
```

## Reproduction

```tsx
function outer(Component: () => React.ReactNode) {
    function Inner() {
        return <Component />;
    }

    return Inner;
}
```

## Actual behavior

Oxlint reports:

```text
Function `Inner` does not capture any variables from its parent scope
```

## Expected behavior

Oxlint should report no violation. `Inner` captures the `Component` parameter. Moving `Inner`
outside `outer` makes `Component` unavailable and changes the program. `eslint-plugin-unicorn@74.0.0`
reports no violation.
